lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2526,6 +2535,33 @@ class StmtEmitter : public EmitterBase,
 
 } // end anonymous namespace
 
+bool isInProceduralRegion(SmallVector<Operation *> &ops) {
+  // First check if any of the ops is in a procedural region.
+  for (auto *op : ops)
+    if (op->getParentOp()->hasTrait<ProceduralRegion>())
+      return true;
+
+  // Also check if any uses of the ops are in a procedural region. This is
+  // necessary because non side-effecting ops are hoisted in PrepareForEmission,
+  // but may be used in a procedural region, and cannot be spilled to wires in
+  // that region.
+  SmallVector<Operation *> worklist(ops);
+  SmallPtrSet<Operation *, 6> seen;
+  while (!worklist.empty()) {
+    auto *op = worklist.pop_back_val();
+    if (seen.contains(op))
+      continue;
+    seen.insert(op);
+    for (auto *user : op->getUsers()) {
+      if (user->hasTrait<ProceduralRegion>())
```

Comment:
user->getParentOp()->hasTrait?

Reply:
This is testing if the use is an ifdef or always block, not that it's in an always block